app/assets/v2/js/grants/detail.js

```diff
@@ -49,36 +71,34 @@ $(document).ready(function() {
     editableFields.forEach(field => disableEdit(field));
   });
 
-  $('#js-cancel_grant').validate({
-    submitHandler: function(form) {
-      var data = {};
-
-      $.each($(form).serializeArray(), function() {
-        data[this.name] = this.value;
-      });
-
-      // let compiledSubscription;
-      //
-      // if (data.contract_version == 0) {
-      //   compiledSubscription = compiledSubscription0;
-      // }
-
-      let deployedSubscription = new web3.eth.Contract(compiledSubscription.abi, data.contract_address);
-
-      web3.eth.getAccounts(function(err, accounts) {
-        deployedSubscription.methods.endContract()
-          .send({from: accounts[0], gasPrice: 4000000000})
-          .on('transactionHash', function() {
-            document.issueURL = document.getElementById('form--input__reference-url').value;
-            enableWaitState('#grants-details');
-          })
-          .on('confirmation', function(confirmationNumber, receipt) {
-            console.log('receipt', receipt);
-            form.submit();
+  $('#cancel_grant').click(function() {
+
+    let contract_address = $('#contract_address').val();
+
+    let deployedSubscription = new web3.eth.Contract(compiledSubscription.abi, contract_address);
+
+    web3.eth.getAccounts(function(err, accounts) {
+      deployedSubscription.methods.endContract()
+        .send({from: accounts[0], gasPrice: 4000000000})
+        .on('transactionHash', function() {
+          enableWaitState('#grants-details');
+        })
+        .on('confirmation', function(confirmationNumber, receipt) {
+          $.ajax({
+            type: 'post',
+            url: '',
+            data: { 'contract_address': contract_address},
```

Comment:
not sure you have access to `contract_address` from this scope

Reply:
It is available. The ajax call inherits from ln76. Just tested again and functionality works.